api/pkg/transformer/feast/transformer_test.go

```diff
@@ -0,0 +1,210 @@
+package feast
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	feast "github.com/feast-dev/feast/sdk/go"
+	"github.com/feast-dev/feast/sdk/go/protos/feast/serving"
+	"github.com/feast-dev/feast/sdk/go/protos/feast/types"
+	"github.com/stretchr/testify/mock"
+	"go.uber.org/zap"
+
+	"github.com/gojek/merlin/pkg/transformer"
+	"github.com/gojek/merlin/pkg/transformer/feast/mocks"
+)
+
+func TestTransformer_Transform(t *testing.T) {
+	logger, _ := zap.NewDevelopment()
+
+	type fields struct {
+		config *transformer.StandardTransformerConfig
+	}
+	type args struct {
+		ctx          context.Context
+		request      []byte
+		feastRequest feast.OnlineFeaturesRequest
+	}
+	type mockFeast struct {
+		request  *feast.OnlineFeaturesRequest
+		response *feast.OnlineFeaturesResponse
+	}
+	tests := []struct {
+		name      string
+		fields    fields
+		args      args
+		mockFeast []mockFeast
+		want      []byte
+		wantErr   bool
+	}{
+		{
+			name: "one config: retrieve one entity, one feature",
+			fields: fields{
+				config: &transformer.StandardTransformerConfig{
+					TransformerConfig: &transformer.TransformerConfig{
+						Feast: []*transformer.FeatureTable{
+							{
+								Project: "default",
+								Entities: []*transformer.Entity{
+									{
+										Name:      "driver_id",
+										ValueType: "string",
+										JsonPath:  "driver_id",
+									},
+								},
+								Features: []*transformer.Feature{
+									{
+										Name:         "driver_trips:average_daily_rides",
+										DefaultValue: 0.0,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				ctx:     context.Background(),
+				request: []byte(`{"driver_id":"1001"}`),
+			},
+			mockFeast: []mockFeast{
+				{
+					request: &feast.OnlineFeaturesRequest{
+						Project: "default", // used as identifier for mocking. must match config
+					},
+					response: &feast.OnlineFeaturesResponse{
+						RawResponse: &serving.GetOnlineFeaturesResponse{
+							FieldValues: []*serving.GetOnlineFeaturesResponse_FieldValues{
+								{
+									Fields: map[string]*types.Value{
+										"driver_trips:average_daily_rides": feast.DoubleVal(1.1),
+									},
+									Statuses: map[string]serving.GetOnlineFeaturesResponse_FieldStatus{
+										"driver_trips:average_daily_rides": serving.GetOnlineFeaturesResponse_PRESENT,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want:    []byte(`{"driver_id":"1001","feast_features":{"driver_id":{"columns":["driver_trips:average_daily_rides"],"data":[1.1]}}}`),
+			wantErr: false,
+		},
+		{
+			name: "two configs: each retrieve one entity, one feature",
```

Comment:
Can we add more test case:
- 2 configs: 2 entities and 2 or more features
- Entity type that's not string
- The entity type in json body is different than the entity type (i.e. integer in the request, but the entity type is string)

And edge cases:
- Json path is incorrect (field is empty or not found)